lang/cs/Org.Apache.REEF.Tang.Tests/Configuration/TestConfiguration.cs

```diff
@@ -425,6 +426,77 @@ public void TestTimerConfigurationWithClassHierarchy()
             timer.sleep();
         }
 
+        [Fact]
+        public void TestListConfig()
+        {
+            IList<string> stringList1 = new List<string>();
+            stringList1.Add("foo");
+            stringList1.Add("bar");
+
+            IList<string> stringList2 = new List<string>();
+            stringList2.Add("test1");
+            stringList2.Add("test2");
+            stringList2.Add("test3");
+            stringList2.Add("test4");
+
+            IConfiguration conf = TangFactory.GetTang().NewConfigurationBuilder()
+                .BindList<ListOfStrings, string>(GenericType<ListOfStrings>.Class, stringList1)
+                .BindList<ListOfStrings2, string>(GenericType<ListOfStrings2>.Class, stringList2)
+                .Build();
+
+            ConfigurationFile.WriteConfigurationFile(conf, "ListOfStringsConf.txt");
+
+            string s = ConfigurationFile.ToConfigurationString(conf);
+            IConfiguration conf2 = ConfigurationFile.GetConfiguration(s);
+
+            ListInjectTest injectTest = (ListInjectTest)TangFactory.GetTang().
+                    NewInjector(conf2).GetInstance(typeof(ListInjectTest));
+
+            Assert.Equal(stringList1, injectTest.List1);
+            Assert.Equal(stringList2, injectTest.List2);
+        }
+
+        [Fact]
+        public void TestListConfigWithAvroSerialization()
+        {
+            IList<string> stringList = new List<string>();
+            stringList.Add("foo");
+            stringList.Add("bar");
+            IConfiguration conf = TangFactory.GetTang().NewConfigurationBuilder()
+                    .BindList<ListOfStrings, string>(GenericType<ListOfStrings>.Class, stringList)
+                    .Build();
+
+            var serializer = new AvroConfigurationSerializer();
+            byte[] bytes = serializer.ToByteArray(conf);
+            IConfiguration conf2 = serializer.FromByteArray(bytes);
+            Assert.Equal(1, conf2.GetBoundList().Count);
+
+            var actualList = conf2.GetBoundList().First().Value;
+            Assert.Equal(stringList, actualList);
+        }
+
+        [Fact]
+        public void TestListSerializeNullStringValue()
```

Comment:
We should probably also have a test that checks for the  empty list, and one that contains empty strings.